dpnp/tests/test_usm_type.py

```diff
@@ -1796,64 +1209,421 @@ def test_nan_to_num(copy, usm_type_a):
+        to_begin = dpnp.array(to_begin, usm_type=usm_type_args)
 
-    res = dp.ediff1d(x, to_end=to_end, to_begin=to_begin)
+    res = dpnp.ediff1d(x, to_end=to_end, to_begin=to_begin)
 
     assert res.usm_type == du.get_coerced_usm_type([usm_type_x, usm_type_args])
 
 
-@pytest.mark.parametrize("usm_type", list_of_usm_types, ids=list_of_usm_types)
+@pytest.mark.parametrize("usm_type", list_of_usm_types)
 def test_unravel_index(usm_type):
-    x = dp.array(2, usm_type=usm_type)
-    result = dp.unravel_index(x, shape=(2, 2))
+    x = dpnp.array(2, usm_type=usm_type)
+    result = dpnp.unravel_index(x, shape=(2, 2))
     for res in result:
         assert res.usm_type == x.usm_type
 
 
-@pytest.mark.parametrize("usm_type", list_of_usm_types, ids=list_of_usm_types)
+@pytest.mark.parametrize("usm_type", list_of_usm_types)
 def test_ravel_index(usm_type):
-    x = dp.array([1, 0], usm_type=usm_type)
-    result = dp.ravel_multi_index(x, (2, 2))
+    x = dpnp.array([1, 0], usm_type=usm_type)
+    result = dpnp.ravel_multi_index(x, (2, 2))
     assert result.usm_type == x.usm_type
 
 
-@pytest.mark.parametrize("usm_type_0", list_of_usm_types, ids=list_of_usm_types)
-@pytest.mark.parametrize("usm_type_1", list_of_usm_types, ids=list_of_usm_types)
+@pytest.mark.parametrize("usm_type_0", list_of_usm_types)
+@pytest.mark.parametrize("usm_type_1", list_of_usm_types)
 def test_ix(usm_type_0, usm_type_1):
-    x0 = dp.array([0, 1], usm_type=usm_type_0)
-    x1 = dp.array([2, 4], usm_type=usm_type_1)
-    ixgrid = dp.ix_(x0, x1)
+    x0 = dpnp.array([0, 1], usm_type=usm_type_0)
+    x1 = dpnp.array([2, 4], usm_type=usm_type_1)
+    ixgrid = dpnp.ix_(x0, x1)
     assert ixgrid[0].usm_type == x0.usm_type
     assert ixgrid[1].usm_type == x1.usm_type
 
 
-@pytest.mark.parametrize("usm_type_x", list_of_usm_types, ids=list_of_usm_types)
-@pytest.mark.parametrize(
-    "usm_type_ind", list_of_usm_types, ids=list_of_usm_types
-)
+@pytest.mark.parametrize("usm_type_x", list_of_usm_types)
+@pytest.mark.parametrize("usm_type_ind", list_of_usm_types)
 def test_choose(usm_type_x, usm_type_ind):
-    chc = dp.arange(5, usm_type=usm_type_x)
-    ind = dp.array([0, 2, 4], usm_type=usm_type_ind)
-    z = dp.choose(ind, chc)
+    chc = dpnp.arange(5, usm_type=usm_type_x)
+    ind = dpnp.array([0, 2, 4], usm_type=usm_type_ind)
+    z = dpnp.choose(ind, chc)
 
     assert chc.usm_type == usm_type_x
     assert ind.usm_type == usm_type_ind
     assert z.usm_type == du.get_coerced_usm_type([usm_type_x, usm_type_ind])
 
 
-# TODO: add it as part of `test_array_creation_from_scratch` function
-@pytest.mark.parametrize("func", ["hamming"])
-@pytest.mark.parametrize("usm_type", list_of_usm_types + [None])
-def test_window(func, usm_type):
-    result = getattr(dp, func)(10, usm_type=usm_type)
-    if usm_type is None:
-        # assert against default USM type
-        assert result.usm_type == "device"
-    else:
+class TestLinAlgebra:
```

Comment:
Should we do it here also?
```suggestion
@pytest.mark.parametrize("usm_type", list_of_usm_types)
class TestLinAlgebra:
```